application/src/main/java/org/opentripplanner/transit/api/request/TripTimeOnDateRequestBuilder.java

```diff
@@ -0,0 +1,132 @@
+package org.opentripplanner.transit.api.request;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.opentripplanner.model.TripTimeOnDate;
+import org.opentripplanner.transit.api.model.FilterValues;
+import org.opentripplanner.transit.model.basic.TransitMode;
+import org.opentripplanner.transit.model.framework.FeedScopedId;
+import org.opentripplanner.transit.model.site.StopLocation;
+import org.opentripplanner.transit.service.ArrivalDeparture;
+
+public class TripTimeOnDateRequestBuilder {
+
+  private static final String INCLUDE_AGENCIES = "includeAgencies";
+  private static final String INCLUDE_ROUTES = "includeRoutes";
+  private static final String EXCLUDE_AGENCIES = "excludeAgencies";
+  private static final String INCLUDE_MODES = "includeModes";
+  private static final String EXCLUDE_ROUTES = "excludeRoutes";
+  private static final String EXCLUDE_MODES = "excludeModes";
+  private final Collection<StopLocation> stopLocations;
```

Comment:
This should be `stopIds` not `StopLocation`

Reply:
I get what you're saying but that means that I look up stops on one end, then take the id, and then in the library code I look up the stop again. Is that ok?

Reply:
We talked about it in the dev meeting and the reason is that many methods in TransitService already use the StopLocation as the input type. Many underlying indices also use it as the key.